src/command_modules/azure-cli-network/azure/cli/command_modules/network/custom.py

```diff
@@ -487,6 +488,138 @@ def update_ag_redirect_configuration(cmd, instance, parent, item_name, redirect_
     return parent
 
 
+def create_ag_rewrite_rule_set(cmd, resource_group_name, application_gateway_name, item_name, no_wait=False):
+    ApplicationGatewayRewriteRuleSet = cmd.get_models(
+        'ApplicationGatewayRewriteRuleSet')
+    ncf = network_client_factory(cmd.cli_ctx).application_gateways
+    ag = ncf.get(resource_group_name, application_gateway_name)
+    new_set = ApplicationGatewayRewriteRuleSet(name=item_name)
+    _upsert(ag, 'rewrite_rule_sets', new_set, 'name')
+    if no_wait:
+        return sdk_no_wait(no_wait, ncf.create_or_update, resource_group_name, application_gateway_name, ag)
+    parent = sdk_no_wait(no_wait, ncf.create_or_update, resource_group_name, application_gateway_name, ag).result()
```

Comment:
Is this necessary? Should we just do:
```suggestion
    parent = ncf.create_or_update(resource_group_name, application_gateway_name, ag)
```

If this is to poll the long running operation, does this behave similarly to our long running operation object / poller?

- [LongRunningOperation](https://github.com/Azure/azure-cli/blob/e0bd4709783e02567a8f9e1b5cfd186514281b9d/src/azure-cli-core/azure/cli/core/commands/__init__.py#L509)
- [LRO usage](https://github.com/Azure/azure-cli/blob/e0bd4709783e02567a8f9e1b5cfd186514281b9d/src/command_modules/azure-cli-vm/azure/cli/command_modules/vm/custom.py#L900-L903)

Reply:
Correct. If you do what you suggest, you will just end up with a poller, which will not work.

Reply:
Yes, but I believe calling LRO on the result of a long running sdk call makes it print "running..." to the screen and adds [other functionalities](https://github.com/Azure/azure-cli/blob/e0bd4709783e02567a8f9e1b5cfd186514281b9d/src/azure-cli-core/azure/cli/core/commands/__init__.py#L585-L634)... 

My suggestion is something like [this](https://github.com/Azure/azure-cli/blob/e0bd4709783e02567a8f9e1b5cfd186514281b9d/src/command_modules/azure-cli-vm/azure/cli/command_modules/vm/custom.py#L900-L903):

```suggestion
    parent_poller = ncf.create_or_update(resource_group_name, application_gateway_name, ag)
    parent = LongRunningOperation(cmd.cli_ctx)(parent_poller)
    return _find_child_item(parent, item_name, path='rewrite_rule_sets', key_path='name')
```